config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch

```diff
@@ -0,0 +1,1818 @@
+From d99da54fd7296a7ac02bed17a7ded0f6096d283c Mon Sep 17 00:00:00 2001
+From: Tomasz Wozniak <t.wozniak@samsung.com>
+Date: Wed, 23 Aug 2017 12:45:48 +0200
+Subject: [PATCH 4/4] Provide working iotjs config for 7.19 with:  -FPU/MPU 
+ -ROMFS, TMPFS  -USB/NSH  -ADC, I2C, SPI, UART, GPIO, TCP/IP, UDP
+
+---
+ configs/stm32f4discovery/usbnshiotjs/Make.defs |  113 ++
+ configs/stm32f4discovery/usbnshiotjs/defconfig | 1589 ++++++++++++++++++++++++
+ configs/stm32f4discovery/usbnshiotjs/setenv.sh |   80 ++
+ 3 files changed, 1782 insertions(+)
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/Make.defs
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/defconfig
+ create mode 100755 configs/stm32f4discovery/usbnshiotjs/setenv.sh
```

Comment:
How about creating files and copying it without applying a patch file like tizenrt?

Reply:
There are too many things in your patch files and it is hard to maintain. If the config file will be changed, how do you manage it? If the version of the nuttx is updated and config file is changed, you will need to delete contents in the patch files and copy text.
The role of your patch will make the file new, not change files. I want to separate it into three files.
And we have copied config file so far. And you should update the documents about the changes.

Reply:
Moved the config file to config/nuttx/stm32f4dis/dot_usbnshiotjs_defconfig.
Removed nuttx patches 1 (network buildbreak fix) and 4 (usbnshiotjs configuration for stm32f4).
Still two patches are left (missing include in string include chain, missing definition for PWM/Timer configuration). Without these nuttx would not build with the provided config.